aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -19,6 +19,49 @@
 namespace at {
 namespace native {
 
+template<class scalar_t>
+void cusolver_LU(int m, int n, scalar_t* dA, int ldda, int* ipiv, int* info);
+
+template<class scalar_t>
+void cusolver_Getri(int n, scalar_t* dA, int ldda, int* ipiv, int* info, const Tensor& ret);
+
+template<>
+void cusolver_LU<double>(int m, int n, double* dA, int ldda, int* ipiv, int* info) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  int lwork;
+  TORCH_CUSOLVER_CHECK(cusolverDnDgetrf_bufferSize(handle, m, n, dA, ldda, &lwork));
+  Tensor buffer = at::empty({lwork}, at::device(at::kCUDA).dtype(at::kDouble));
+  Tensor devInfo = at::empty({1}, at::device(at::kCUDA).dtype(at::kInt));
+  TORCH_CUSOLVER_CHECK(cusolverDnDgetrf(handle, m, n, dA, ldda, buffer.data_ptr<double>(), ipiv, devInfo.data_ptr<int>()));
+  *info = devInfo.item<int>();
+}
+
+template<>
+void cusolver_LU<float>(int m, int n, float* dA, int ldda, int* ipiv, int* info) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  int lwork;
+  TORCH_CUSOLVER_CHECK(cusolverDnSgetrf_bufferSize(handle, m, n, dA, ldda, &lwork));
+  Tensor buffer = at::empty({lwork}, at::device(at::kCUDA).dtype(at::kFloat));
+  Tensor devInfo = at::empty({1}, at::device(at::kCUDA).dtype(at::kInt));
+  TORCH_CUSOLVER_CHECK(cusolverDnSgetrf(handle, m, n, dA, ldda, buffer.data_ptr<float>(), ipiv, devInfo.data_ptr<int>()));
+  *info = devInfo.item<int>();
+}
+
+template<>
+void cusolver_Getri<double>(int n, double* dA, int ldda, int* ipiv, int* info, const Tensor& ret) {
```

Comment:
Thanks! I think the purpose of this function is "getri" which finds the "inverse". The reason we use cusolverXgetrs is because cusolver doesn't have "getri" at this moment. So we let rhs of AX=B be identity matrix for this. It's not a general "getrs" wrapper.

Reply:
I think you are right. They are actually getrs. Will rename.